### DISCUSSION TITLE
Collection Vs Environment

### DISCUSSION THREAD

**Discussion by loranger** [usebruno/bruno#1551] [Q&A]
Can someone explain the differences between Collection and Environment, please?

I'd like to understand the pros and cons, especially as it looks like you can't re-use variables in queries: I've moved my API key from environment to collection parameters and my authentication variable remains red

**Comment by mdreano** [comment]
I see collections as a set of endpoints. (one collection per application for me)
Environments are a set of variables i use when calling those endpoints (including some part of the endpoint path)
